scrapy/core/engine.py

```diff
@@ -73,12 +74,22 @@ def __init__(self, crawler, spider_closed_callback: Callable) -> None:
         self.spider: Optional[Spider] = None
         self.running = False
         self.paused = False
-        self.scheduler_cls = load_object(crawler.settings["SCHEDULER"])
+        self.scheduler_cls = self._get_scheduler_class(crawler.settings)
         downloader_cls = load_object(self.settings['DOWNLOADER'])
         self.downloader = downloader_cls(crawler)
         self.scraper = Scraper(crawler)
         self._spider_closed_callback = spider_closed_callback
 
+    def _get_scheduler_class(self, settings: BaseSettings) -> type:
+        from scrapy.core.scheduler import BaseScheduler
+        scheduler_cls = load_object(settings["SCHEDULER"])
+        if not issubclass(scheduler_cls, BaseScheduler):
+            raise TypeError(
+                f"The provided scheduler class ({settings['SCHEDULER']})"
+                " does not fully implement the scheduler interface"
+            )
```

Comment:
This will break any existing project with a custom scheduler class :sweat_smile:

Reply:
It shouldn't, the subclass check is done via a [metaclass](https://github.com/scrapy/scrapy/pull/3559/files#diff-e957998ce47b270d8ae072df61f91368d94a20bb7bff60c337a0900eb192bf6dR27), it makes sure the necessary methods exist.

Reply:
Cool!